projects/laji/src/app/+news/news.component.ts

```diff
@@ -1,64 +1,70 @@
 import { HeaderService } from '../shared/service/header.service';
 import { Title } from '@angular/platform-browser';
+import { NewsFacade } from './news.facade';
 
 
 @Component({
   selector: 'laji-news',
   templateUrl: './news.component.html',
-  styleUrls: ['./news.component.css'],
+  styleUrls: ['../../styles/information.scss', './news.component.css'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class NewsComponent implements OnInit, OnDestroy {
+export class NewsComponent implements OnInit {
   public newsItem$: Observable<News>;
-  private subTrans: Subscription;
 
   constructor(private route: ActivatedRoute,
-              private newsService: NewsService,
-              private cd: ChangeDetectorRef,
-              private logger: Logger,
-              private store: NewsStore,
+              private newsFacade: NewsFacade,
               private headerService: HeaderService,
               private title: Title
   ) {}
 
   ngOnInit() {
-    this.newsItem$ = this.store.state$.pipe(
-      map(state => state.current),
-      distinctUntilChanged(),
-      filter(info => !!info),
-      tap(info => {
-        this.title.setTitle(info.title + ' | ' + this.title.getTitle());
-      })
-    );
-    this.subTrans = this.route.params.pipe(
+    this.newsItem$ = this.route.params.pipe(
       map(params => params['id']),
-      switchMap(id => this.store.state.current && this.store.state.current.id === id ?
-        ObservableOf(this.store.state.current) : this.newsService.get(id))
-    ).subscribe(newsItem => {
-      this.store.setCurrent(newsItem);
-      if (newsItem.featuredImage) {
-        this.headerService.updateFeatureImage(newsItem.featuredImage);
-      }
-      setTimeout(() => {
-        const paragraph = (document.getElementById('wrapper'))?.getElementsByTagName('p')?.item(0)?.innerText || newsItem.content.replace(/<[^>]*>?/gm, '')
-        .substring(0, 180).replace(/^(.{160}[^\s]*).*/, '$1');
-        this.headerService.updateMetaDescription(paragraph);
-        this.headerService.createTwitterCard(newsItem.title + ' | ' + this.title.getTitle());
-      }, 0);
-    });
+      tap(id => this.newsFacade.activate(id)),
+      switchMap((id) => this.newsFacade.active$.pipe(
+        filter(news => news?.id === id),
+      )),
+      delay(0),
+      tap(news => this.updateHeaders(news))
+    );
+  }
+
+  private updateHeaders(news: News): void {
+    const pageTitle = news.title + ' | ' + this.title.getTitle();
+    const paragraph = this.getNewsDescription(news);
+
+    this.title.setTitle(pageTitle);
+    this.headerService.createTwitterCard(pageTitle);
+    this.headerService.updateMetaDescription(paragraph);
+
+    if (news.featuredImage) {
+      this.headerService.updateFeatureImage(news.featuredImage);
+    }
   }
 
-  ngOnDestroy() {
-    if (this.subTrans) {
-      this.subTrans.unsubscribe();
+  private getNewsDescription(news: News): string {
```

Comment:
This is the same algorithm that #12 introduced into header service? Could we use that instead of duplicate code?

Reply:
Yes as it's states in the #12 commit message :)
> When this is merged #10 can be refined to use HeaderService.getDescription and the whole deprecated store can be removed